src/rulesets/LinearAlgebra/dense.jl

```diff
@@ -40,17 +40,17 @@ end
 ##### `det`
 #####
 
-function frule((_, ẋ), ::typeof(det), x)
+function frule((_, ẋ), ::typeof(det), x::Union{Number, AbstractMatrix})
```

Comment:
why the type annotations? (these do i believe cover all `det`/`logdet` methods in Base, but now not those extended by other packages, i suppose?) 

dose it warrant a comment in the code (or docs?) ?

Reply:
Correct, I do not thing it deserves an annoation though,
it should be our practice to always include these when the primal method has them

Reply:
https://github.com/JuliaDiff/ChainRulesCore.jl/issues/155